### PR TITLE
[Bugfix] Sentry Issue Type Error in programming submission service

### PR DESCRIPTION
**Pull request by lschlesinger**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, editor, instructor, admin) on the test server https://artemistest.ase.in.tum.de. 

### Motivation and Context

This PR solves the [Sentry Issue ARTMIS-26](https://sentry.ase.in.tum.de/organizations/artemis/issues/71/?project=2)

### Description

Error: TypeError call(src/main/webapp/app/exercises/programming/participate/programming-submission.service)
Cannot read property 'next' of undefined.
I inspected the use of `.next()` at other places in the respective service and adapted two occurrences of `.next()` without prior null check.

### Steps for Testing

As I could not reproduce the issue, I would say following procedure is enough to test the changes:

-> automatic assessment

1. Log in to Artemis
2. Search a programming exercise without due date and manual correction enabled
3. As a student, participate in that programming exercise  and submit a solution 
4. inspect console log, are there any TypeErrors? Is the automatically created result as expected?
5. As a tutor/instructor, assess the submission and save the assessment

-> manual assessment

1. Log in to Artemis
2. Search a programming exercise with due date and manual correction enabled
3. As a student, participate in that programming exercise and submit a solution 
4. Change the due date to a date in the past (or wait if the due date is within the next minutes)
5. inspect console log, are there any TypeErrors? Is the automatically created result as expected?
6. As a tutor/instructor, assess the submission and save the assessment

### CODE REVIEW

File: src/main/webapp/app/exercises/programming/participate/programming-submission.service.ts [hunk 1]
```diff
@@ -118,7 +118,9 @@ export class ProgrammingSubmissionService implements IProgrammingSubmissionServi
 
         // For every 100 submissions, we increase the expected time by 1 minute.
         this.currentExpectedResultETA = this.DEFAULT_EXPECTED_RESULT_ETA + Math.floor(buildingSubmissionCount / 100) * 4000 * 60;
-        this.resultEtaSubject.next(this.currentExpectedResultETA);
+        if (this.resultEtaSubject) {
+            this.resultEtaSubject.next(this.currentExpectedResultETA);
+        }
```
Comment (jpbernius) [inline src/main/webapp/app/exercises/programming/participate/programming-submission.service.ts:123]:
```suggestion
        this.resultEtaSubject?.next(this.currentExpectedResultETA);
```

File: src/main/webapp/app/exercises/programming/participate/programming-submission.service.ts [hunk 2]
```diff
@@ -450,7 +452,10 @@ export class ProgrammingSubmissionService implements IProgrammingSubmissionServi
             )
             .subscribe((exerciseBuildState: ExerciseSubmissionState) => {
                 this.exerciseBuildState = { ...this.exerciseBuildState, [exerciseId]: exerciseBuildState };
-                this.exerciseBuildStateSubjects.get(exerciseId)!.next(exerciseBuildState);
+                const exerciseBuildStageSubject = this.exerciseBuildStateSubjects.get(exerciseId);
+                if (exerciseBuildStageSubject) {
+                    exerciseBuildStageSubject.next(exerciseBuildState);
+                }
```
Comment (jpbernius) [inline src/main/webapp/app/exercises/programming/participate/programming-submission.service.ts:458]:
```suggestion
                this.exerciseBuildStateSubjects.get(exerciseId)?.next(exerciseBuildState);
```

**Review by jpbernius — CHANGES_REQUESTED** [review]
Could this also work with optional chaining?

**Review by ga62suz — CHANGES_REQUESTED** [review]
Tested on TS2.

With manual assessment enabled + no due date, students don't get an immediate build + test when they submit after editing submission with online editor.
Compared to TS1 (deployed with another branch at the time) where build trigger was instant.

**Review by Olimbia23 — APPROVED** [review]
Looks good 👍

**Review by rs-maker — APPROVED** [review]
Good fix.

**Review by SimonEntholzer — APPROVED** [review]
Changes look good and fix the issue :+1:

**Review by ivohashamov — APPROVED** [review]
Changes look good

**Review by FrankeLukas — APPROVED** [review]
Lgtm. @ga62suz TS2 uses a different setup compared TS1, so I don't think delays in the builds are caused by this PR.

**Pull request merged**